src/globus_cli/exception_handling/hooks.py

```diff
@@ -25,17 +25,31 @@ def _pretty_json(data: dict, compact=False) -> str:
     exit_status=4,
 )
 def handle_internal_auth_requirements(exception: CLIAuthRequirementsError) -> None:
-    if not (exception.gare and exception.gare.authorization_parameters.required_scopes):
+    gare = exception.gare
+    if not gare:
         click.secho(
             "Fatal Error: Unsupported internal auth requirements error!",
             bold=True,
             fg="red",
         )
         click.get_current_context().exit(255)
```

Comment:
Is exiting with 255 significant compared to exiting with some other value?

In the past I've used firmware manipulation tools that had dozens of possible error codes that each conveyed documented, nuanced information; we're not doing anything like that, but if there's a convention to use 255 for fatal errors and, say, 1 for regular ol' "didn't work" errors, I want to know that convention.

Not related to this review but wanted to surface my thoughts since I'm seeing this here.

Reply:
We have some defined and documented exit codes.
1 for "normal failure" (e.g. request failure)
2 for argparsing (typical)
3 for user-facing entity type mismatches (e.g. storage-gateway list invoked on a GCP endpoint)
4 for session errors (note `exit_status=4` in the decorator)

We also reserve the range of exit statuses 50-99 for user-use, via `--map-http-status`. (You are also allowed to `--map-http-status '404=0'`.)

255 is not explicitly documented anywhere, but is used for internal/fatal errors, to ensure that in the event that we encounter a condition like this, we can exit with a status other than the ones which we're already using.

We can add it to the exit code documentation which we render on the docs site (e.g. [one command's exit code doc](https://docs.globus.org/cli/reference/api_auth/#exit_status)). I'm not sure I want it clogging up `--help` output everywhere.

Reply:
No, not something to add to `--help` output everywhere. I just wanted to understand the conventions (and whether they're documented).

Might be good to establish constants in the future so they're used throughout.

Reply:
Yeah, I'm not averse to a follow-up which puts these all into a file like `globus_cli.constants` and then we use `exit(ECODE_FATAL_ERROR)` or something like that. That way the convention is codified in the code without us having to "just know" (the current state of the art).